### DISCUSSION TITLE
并发执行SQL执行耗时过高，想知道是为啥

### DISCUSSION THREAD

**Discussion by Accelarator** [apache/doris#14365] [A - General / Q&A]
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:w="urn:schemas-microsoft-com:office:word"
xmlns:m="http://schemas.microsoft.com/office/2004/12/omml"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Word.Document>
<meta name=Generator content="Microsoft Word 15">
<meta name=Originator content="Microsoft Word 15">
<link rel=File-List
href="file:///C:/Users/tt/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">

<link rel=themeData
href="file:///C:/Users/tt/AppData/Local/Temp/msohtmlclip1/01/clip_themedata.thmx">
<link rel=colorSchemeMapping
href="file:///C:/Users/tt/AppData/Local/Temp/msohtmlclip1/01/clip_colorschememapping.xml">

<style>

</style>

</head>

<body lang=ZH-CN style='tab-interval:21.0pt;word-wrap:break-word;text-justify-trim:
punctuation'>

<p class=MsoNormal align=left style='text-align:left;mso-pagination:widow-orphan'><span
style='font-size:16.0pt;font-family:"微软雅黑",sans-serif;mso-bidi-font-family:
宋体;mso-font-kerning:0pt'>执行<span lang=EN-US>SQL</span>：大表（<span lang=EN-US>7</span>亿）<span
lang=EN-US>join</span>小表（<span lang=EN-US>400w</span>）<span lang=EN-US><o:p></o:p></span></span></p>

<p class=MsoNormal align=left style='text-align:left;mso-pagination:widow-orphan'><span
style='font-size:16.0pt;font-family:"微软雅黑",sans-serif;mso-bidi-font-family:
宋体;mso-font-kerning:0pt'>起不同的线程数同时执行相同<span lang=EN-US>SQL</span>的耗时如下<span
lang=EN-US><o:p></o:p></span></span></p>

线程数 | 执行时间（秒）
-- | --
1 | 2.5
5 | 13
10 | 22
20 | 45
30 | 65
40 | 94
50 | 102

</body>

</html>

并发数统一为set parallel_fragment_exec_instance_num = 8;
以50个线程数的profile为例：
![image](https://user-images.githubusercontent.com/18416349/202398426-6619f0c8-d472-4769-a487-8845f27151f7.png)
![image](https://user-images.githubusercontent.com/18416349/202398521-2bc828c7-850e-4528-9931-83653b6a580b.png)
![image](https://user-images.githubusercontent.com/18416349/202398605-535a88a0-fbb2-4433-a3d0-7392b942749e.png)

**Comment by Accelarator** [comment]
想知道瓶颈在哪，要怎么排查

**Reply by kpfly** [reply]
Is the CPU resource a bottleneck?

**Reply by Accelarator** [reply]
> Is the CPU resource a bottleneck?

是的，我们的机器是32core，与其他hadoop组件混合部署。当10并发时CPU就达到23core了，所以初步可判断是CPU达到瓶颈

**Reply by kpfly** [reply]
You should add more servers

**Comment by liutang123** [comment]
`ScannerWorkerWaitTime` is hight, maybe you can try increse `doris_scanner_thread_pool_thread_num`.

**Reply by Accelarator** [reply]
> `ScannerWorkerWaitTime` is hight, maybe you can try increse `doris_scanner_thread_pool_thread_num`.

设置了```doris_scanner_thread_pool_thread_num=256```后发现时间没变化，后来发现是CPU达到瓶颈了，后续不知道咋优化了

**Reply by dataalive** [reply]
cpu都满了有瓶颈了，你想要的效果是什么呢？

**Reply by kpfly** [reply]
> > `ScannerWorkerWaitTime` is hight, maybe you can try increse `doris_scanner_thread_pool_thread_num`.
> 
> 设置了`doris_scanner_thread_pool_thread_num=256`后发现时间没变化，后来发现是CPU达到瓶颈了，后续不知道咋优化了

需要加机器或者增加机器配置了
